src/vma/proto/dst_entry_udp_mc.cpp

```diff
@@ -81,6 +81,24 @@ bool dst_entry_udp_mc::conf_l2_hdr_and_snd_wqe_ib()
 	return ret_val;
 }
 
+void dst_entry_udp_mc::set_src_addr()
+{
+	m_pkt_src_ip = INADDR_ANY;
+	
+	if (m_bound_ip) {
+		m_pkt_src_ip = m_bound_ip;
```

Comment:
Described case is already handled in UC since in dst_entry::set_src_addr() we set m_pkt_src_ip to m_route_src_ip which will include m_bound_ip. However, in MC we don't update m_route_src_ip since we have different implementation for resolve_net_dev in dst_entry_udp_mc class and as result we used m_bound_ip directly.